File: domain/src/main/kotlin/com/sidequest/domain/Domain.kt
package com.sidequest.domain

/**
 * The :domain module holds the client's pure, portable logic (classification,
 * board aggregation, validation, completion counting, due-set computation,
 * games scoring/generation, leaderboard aggregation, sync conflict resolution).
 *
 * It has no Android dependencies so the same logic can be reused and validated
 * with the shared Correctness Properties. Concrete types and logic are added in
 * subsequent tasks (domain models, classification, etc.).
 */
internal object Domain

File: ios/SideQuestKit/Sources/SideQuestKit/Domain/Domain.swift
import Foundation

// MARK: - Portable Domain Logic (placeholder)
//
// This folder will hold the pure Swift domain logic (no I/O) that MUST behave
// identically to the Android client and the backend (Req 3): board
// aggregation/ordering, completion counting, timeframe/due-set resolution,
// bucket-name validation, action-plan progress/reorder, and sync conflict
// resolution (last-writer-wins). This is the layer governed by the shared
// Correctness Properties.
//
// Implemented by task 4 ("Implement the portable domain logic"). This
// placeholder establishes the folder in the shared module so the property
// tests (SwiftCheck) can target it.

/// Marker namespace for the shared portable domain logic. Replaced by the real
/// pure functions/types in task 4.
public enum Domain {}
